src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final Random random = new Random();
+    private static final Shape[] shapes = Shape.values();
+
+    public Figure getRandomFigure() {
+        int randomNumber = random.nextInt(shapes.length);
+        String figureName = shapes[randomNumber].name();
+
+        switch (figureName) {
+            case "CIRCLE":
+                return new Circle();
```

Comment:
```suggestion
                return getRandomCircle();
```
lets add private methods to create each random figure, set all fields in those methods - not in constructors